presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestMergeWindows.java

```diff
@@ -199,15 +199,44 @@ public void testIdenticalWindowSpecificationsABcpA()
                 anyTree(
                         window(windowMatcherBuilder -> windowMatcherBuilder
                                         .specification(specificationA)
-                                        .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(DISCOUNT_ALIAS))),
+                                        .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(DISCOUNT_ALIAS)))
+                                        .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(QUANTITY_ALIAS))),
                                 window(windowMatcherBuilder -> windowMatcherBuilder
                                                 .specification(specificationB)
                                                 .addFunction(functionCall("lag", COMMON_FRAME, ImmutableList.of(QUANTITY_ALIAS, "ONE", "ZERO"))),
                                         project(ImmutableMap.of("ONE", expression("CAST(1 AS bigint)"), "ZERO", expression("0.0")),
-                                                window(windowMatcherBuilder -> windowMatcherBuilder
-                                                                .specification(specificationA)
-                                                                .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(QUANTITY_ALIAS))),
-                                                        LINEITEM_TABLESCAN_DOQSS))))));
+                                                LINEITEM_TABLESCAN_DOQSS)))));
+    }
+
+    @Test
+    public void testIdenticalWindowSpecificationsABfilterA()
```

Comment:
Could you please add a comment here to explain why this test is important?
It may not be obvious why this query is interesting, especially that it's connected to extra nodes which are going to be gotten rid of using `pullUnaryNodeAboveProjects`.